nixos/modules/services/misc/gitlab.nix

```diff
@@ -774,6 +775,10 @@ in {
 
           ${pkgs.openssl}/bin/openssl rand -hex 32 > ${cfg.statePath}/gitlab_shell_secret
```

Comment:
I chowned `gitlab_shell_secret` back to `root:root` and it breaks here.
` 4yjg33mxs50ngydn1glpb9zcx15i7np1-unit-script-gitlab-pre-start[24046]: /nix/store/4yjg33mxs50ngydn1glpb9zcx15i7np1-unit-script-gitlab-pre-start: line 14: /var/www/gitlab/gitlab_shell_secret: Permission denied`

Reply:
Yes, this hinges on systemd-tmpfiles getting to do its job before the script starts, which it should on reboot or when the configuration is activated, if I'm not mistaken. You can manually try it with `systemd-tmpfiles --create`.